src/c++/perf_analyzer/test_dataloader.cc

```diff
@@ -484,6 +593,52 @@ TEST_CASE(
   CHECK_EQ(shape[1], 2);
 }
 
+TEST_CASE(
+    "dataloader: ParseData: Multiple Calls simple" *
+    doctest::description(
+        "ParseData can be called multiple times. The data should "
+        "accumulate in stream 0 when no nested arrays"))
+{
+  std::string json_str1{R"({"data": [{ "INPUT1": [1] }]})"};
+  std::string json_str2{R"({"data": [{ "INPUT1": [2] }]})"};
+  std::string json_str3{R"({"data": [{ "INPUT1": [3] }]})"};
+
+  MockDataLoader dataloader;
+  std::shared_ptr<ModelTensorMap> inputs = std::make_shared<ModelTensorMap>();
+  std::shared_ptr<ModelTensorMap> outputs = std::make_shared<ModelTensorMap>();
+
+  ModelTensor input1 = TestDataLoader::CreateTensor("INPUT1");
+
+  inputs->insert(std::make_pair(input1.name_, input1));
+
+  cb::Error status = dataloader.ReadDataFromStr(json_str1, inputs, outputs);
+  REQUIRE(status.IsOk());
+  CHECK_EQ(dataloader.GetDataStreamsCount(), 1);
+  CHECK_EQ(dataloader.GetTotalSteps(0), 1);
+
+  status = dataloader.ReadDataFromStr(json_str2, inputs, outputs);
+  REQUIRE(status.IsOk());
+  CHECK_EQ(dataloader.GetDataStreamsCount(), 1);
+  CHECK_EQ(dataloader.GetTotalSteps(0), 2);
+
+  status = dataloader.ReadDataFromStr(json_str3, inputs, outputs);
+  REQUIRE(status.IsOk());
+  CHECK_EQ(dataloader.GetDataStreamsCount(), 1);
+  CHECK_EQ(dataloader.GetTotalSteps(0), 3);
+
+  // Confirm the correct data is in the dataloader
+  //
+  const uint8_t* data_ptr{nullptr};
+  size_t batch1_size;
+
+  dataloader.GetInputData(input1, 0, 2, &data_ptr, &batch1_size);
+
+  const int32_t* input_data = reinterpret_cast<const int32_t*>(data_ptr);
+  CHECK_EQ(input_data[0], 3);
+  CHECK_EQ(batch1_size, 4);
+}
+
+
```

Comment:
What is the coverage level after these tests?

Reply:
There is one line that I determined can't be hit. Otherwise it is basically hitting 100% of the testable code (other than some I/O)